# Type: feat
# Subtype: E1
# English Summary: Add validation for intentionElement
# Chinese Summary: 添加对intentionElement相关校验
# Entry Number: 30

# === Git Diff Content ===

diff --git a/platform/lang-impl/src/com/intellij/codeInsight/daemon/impl/ShowIntentionsPass.java b/platform/lang-impl/src/com/intellij/codeInsight/daemon/impl/ShowIntentionsPass.java
index a40300a..394fdde 100644
--- a/platform/lang-impl/src/com/intellij/codeInsight/daemon/impl/ShowIntentionsPass.java
+++ b/platform/lang-impl/src/com/intellij/codeInsight/daemon/impl/ShowIntentionsPass.java
@@ -325,7 +325,9 @@ public class ShowIntentionsPass extends TextEditorHighlightingPass {
           intentionOffset = offset - 1;
         }
       }
-      collectIntentionsFromDoNotShowLeveledInspections(project, hostFile, intentionElement, intentionOffset, intentions);
+      if (intentionElement != null && intentionElement.getManager().isInProject(intentionElement)) {
+        collectIntentionsFromDoNotShowLeveledInspections(project, hostFile, intentionElement, intentionOffset, intentions);
+      }
     }
 
     final int line = hostDocument.getLineNumber(offset);
